components/brave_vpn/resources/panel/components/toggle/index.tsx

```diff
@@ -24,41 +24,93 @@ interface ToggleProps {
   disabled: boolean
 }
 
-function Toggle (props: ToggleProps) {
+function Toggle(props: ToggleProps) {
   const dispatch = useDispatch()
   const isOn = useIsOnSelector()
   const status = useSelector(state => state.connectionStatus)
 
-  const handleToggleChange = (isOn: boolean) => {
-    if (isOn) dispatch(Actions.connect())
+  const handleToggleChange = ({ checked }: { checked: boolean }) => {
+    if (checked) dispatch(Actions.connect())
     else dispatch(Actions.disconnect())
   }
 
+  let statusTextColor : string = color.text.secondary
+  if (status === ConnectionState.CONNECTED) {
+    statusTextColor = color.systemfeedback.successText
+  } else if (status === ConnectionState.CONNECTING ||
+             status === ConnectionState.DISCONNECTING) {
+    statusTextColor = color.text.interactive
+  }
+
+  let icon = 'remove-circle-filled'
+  if (status === ConnectionState.CONNECTED) {
+    icon = 'check-circle-filled'
+  }
+
+  const getStatusIndicator = () => {
```

Comment:
optional:

we can lift these two functions out of the component and have them take a `status` argument. It'll simplify this component and the function won't need to be recreated every rerender